addons/survey/models/survey_question.py

```diff
@@ -147,11 +147,11 @@ def default_get(self, fields):
     triggering_question_id = fields.Many2one(
         'survey.question', string="Triggering Question", copy=False, compute="_compute_triggering_question_id",
         store=True, readonly=False, help="Question containing the triggering answer to display the current question.",
-        domain="""[('survey_id', '=', survey_id),
-                 '&', ('question_type', 'in', ['simple_choice', 'multiple_choice']),
-                 '|',
-                     ('sequence', '<', sequence),
-                     '&', ('sequence', '=', sequence), ('id', '<', id)]""")
+        domain="[('survey_id', '=', survey_id), \
```

Comment:
if it's just to improve the help of the field, you can do in master (it's not really a fix)